File: backend/app/schemas/gbi.py
from typing import Any

from pydantic import BaseModel


class GbiNode(BaseModel):
    id: str
    name: str
    children: list["GbiNode"] | None = None


class GbiHierarchy(BaseModel):
    root: GbiNode | None = None


class GbiAssignProjectsRequest(BaseModel):
    gbi_id: str
    project_ids: list[str]


class GbiUnassignProjectsRequest(BaseModel):
    project_ids: list[str]

File: frontend/src/types/gbi.ts
export interface GbiNode {
  id: string
  name: string
  children?: GbiNode[]
}

export interface GbiHierarchy {
  root: GbiNode | null
}

export interface GbiCloudLeadUser {
  id: string
  name: string
  email: string
  department: string
  team?: string
  initials: string
  role: string[]
  gbi_ids: string[]
}
